R/mzQCjson.R

```diff
@@ -0,0 +1,255 @@
+library(stringr)
+library(readr)
+
+#'
+#' Get the information of each cv Term from a obo file.
+#' 
+#' 
+#' 
+#' @param file "xxx.obo"
+#' @return A list containing cv Term's information
+#' 
+#' @export
+#' 
+#' 
+#' 
+read_cvTerm <- function(file){
+  obo <- data.frame(read.csv(file = file, header = FALSE))
+  cv_Term <- list(cvRef = list(),accession= list(), name = list(), units = list())
+  term_index <- 0
+  for (x in obo$V1) {
+    if(x == "[Term]"){
+      term_index = term_index + 1
+      cv_Term$units[term_index] = ""
+    }
+    if(substring(x,1,2)  == "id"){
+      cv_Term$cvRef[term_index] = substring(str_sub(x, str_locate(x, ":")[1]+2), 1,2)
+      cv_Term$accession[term_index] = str_sub(x, str_locate(x, ":")[1]+2)
+    }
+    if(substring(x,1,4)  == "name"){
+      cv_Term$name[term_index] = str_sub(x, str_locate(x, ":")[1]+2)
+
+    }
+    if(substring(x,1,4)  == "is_a"){
+      ## Connect multiple unit through commas
+      cv_Term$units[term_index] = paste(cv_Term$units[term_index], str_sub(x, str_locate(x, ":")[1]+2), seq=",")
```

Comment:
simply append a list item as computed here https://github.com/cbielow/PTXQC/pull/84/files#diff-3fdc955b9f99affb4c9834016d34a9f2R45

Then you can get rid of the whole block below